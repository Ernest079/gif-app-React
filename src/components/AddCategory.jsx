import { useState } from "react";

export const AddCategory = ({addCategory}) => {
  const [inputValue, setInputValue] = useState('Naruto');
  const onInputChange = ({target}) =>{
    setInputValue(target.value);
  }
  const onSubmit = (event) => {
    event.preventDefault();
    if(inputValue.trim().length <=1){
      return;
    }
    addCategory(cat => [inputValue,...cat] );
    setInputValue('');
  }
  return (
    <form onSubmit={onSubmit}>
      
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={(event) => onInputChange(event)}
      />

    </form>
  );
}
