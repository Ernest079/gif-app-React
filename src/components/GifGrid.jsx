import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";


export const GifGrid = ({category}) => {

  const [images, setimages] = useState([]);

  const getImages = async() => {
    const newImages = await getGifs(category);
    setimages(newImages);
  }

  useEffect(() => {
    getImages();    
  },[]);

  return (
    <>
      <h4>
        {category}
      </h4>
      {images.map((image) => (
        <div key={image.id}>
          <ol>{image.title}</ol>
        </div>
      ))}
    </>
  )
}
