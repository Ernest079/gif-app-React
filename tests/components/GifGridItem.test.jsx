import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";

describe('Pruebas en GifGridItem', () => {
  const title = "Haikyuu";
  const url = "https://haykyuu.gif/";
  test('Debe hacer match con el Snapshot', () => {
    const {container}=render(
      <GifGridItem
        title = {title}
        url = {url}
      />
    );

    expect(container).toMatchSnapshot();
  });

  test('Debe de mostrar la imagen con el url y el alr indicado', () => {
    render(
      <GifGridItem
        title = {title}
        url = {url}
      />
    );

    // expect(screen.getByRole('img').src).toBeTruthy();
    const {src, alt} = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Debe de mostrar el titulo en el componente', () => {
    render(
      <GifGridItem
        title = {title}
        url = {url}
      />
    );
    expect(screen.getByText(title)).toBeTruthy();
  });
});