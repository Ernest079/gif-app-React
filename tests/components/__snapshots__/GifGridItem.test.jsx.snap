// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe hacer match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Haikyuu"
      src="https://haykyuu.gif/"
    />
    <p>
      Haikyuu
    </p>
  </div>
</div>
`;
